cmd/nerdctl/container_run.go

```diff
@@ -733,16 +721,6 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 			if errE := os.RemoveAll(stateDir); errE != nil {
 				isErr = true
 			}
-			if name != "" {
-				var errE error
-				if containerNameStore, errE = namestore.New(dataStore, globalOptions.Namespace); errE != nil {
-					isErr = true
-				}
-				if errE = containerNameStore.Release(name, id); errE != nil {
-					isErr = true
-				}
-
-			}
```

Comment:
The name has to be reserved as soon as the container is createad, so this code shouldn't be moved

Reply:
ping @AkihiroSuda

Reply:
Sorry, my comment is for L690 `if err := containerNameStore.Acquire(name, id); err != nil {`

Reply:
Fixed

Reply:
> The name has to be reserved as soon as the container is created, so this code shouldn't be moved

this code shouldn't be removed too, as failed creation in this level need a cleanup too